### PR TITLE
Merge from istio:master, change install source to istio-install and setup istioctl

### PR DESCRIPTION
**Pull request by yutongz**

### CODE REVIEW

File: tests/e2e/util/kubeUtils.go [hunk 1]
```diff
@@ -94,3 +100,31 @@ func GetIngress(n string) (string, error) {
 	err := fmt.Errorf("cannot get ingress")
 	return "", err
 }
+
```
Comment (sebastienvas) [inline tests/e2e/util/kubeUtils.go]: please move to istioUtils.go

File: kubernetes/istio-install/istio-ingress.yaml [hunk 2]
```diff
@@ -21,11 +21,6 @@ kind: ServiceAccount
 metadata:
   name: istio-ingress-service-account
 ---
-apiVersion: v1
```
Comment (yutongz) [inline kubernetes/istio-install/istio-ingress.yaml]: merge issue, this part is replicated

File: tests/e2e/framework/kubernetes.go [hunk 3]
```diff
@@ -43,6 +45,12 @@ var (
 	caHub      = flag.String("ca_hub", "", "Ca hub")
 	caTag      = flag.String("ca_tag", "", "Ca tag")
 	verbose    = flag.Bool("verbose", false, "Debug level noise from proxies")
+
+	modules = []string{
```
Comment (yutongz) [inline tests/e2e/framework/kubernetes.go]: I think we can keep using istio-install for now just following kubeTest.sh. We can move to istio-auth-16 later.

**Comment by yutongz** [comment]
PTAL

File: tests/e2e/framework/istioctl.go [hunk 4]
```diff
@@ -31,6 +34,7 @@ const (
 
 var (
 	remotePath = flag.String("istioctl_url", os.Getenv(istioctlURL), "URL to download istioctl")
+	pfPID      int
```
Comment (sebastienvas) [inline tests/e2e/framework/istioctl.go]: I would move pfPID in the struct, and store the process directly instead of store the pid, that way you can kill directly the process like we saw with Dan.

File: tests/e2e/framework/kubernetes.go [hunk 5]
```diff
@@ -97,39 +97,43 @@ func newKubeInfo(tmpDir, runID string) *KubeInfo {
...
 // Setup set up Kubernetes prerequest for tests
 func (k *KubeInfo) Setup() error {
-	if err := os.Mkdir(k.yamlDir, os.ModeDir|os.ModePerm); err != nil {
+	var err error
+	if err = os.Mkdir(k.yamlDir, os.ModeDir|os.ModePerm); err != nil {
 		return err
 	}
-	if err := k.Istioctl.Install(); err != nil {
+
+	if err = k.Istioctl.Install(); err != nil {
```
Comment (yutongz) [inline tests/e2e/framework/kubernetes.go]: It's kind of tricky, since deploying app requires both istioctl and "ISTIO_MANAGER_ADDRESS", while on the other hand, setting up istioctl have to wait manager pod.  As a solution, I change apps to appManager and make it cleanable too. And move appDeploy to appManager.Setup()

File: tests/e2e/framework/kubernetes.go [hunk 6]
```diff
@@ -97,21 +100,19 @@ func newKubeInfo(tmpDir, runID string) *KubeInfo {
 		Verbosity:  verbosity,
 		TmpDir:     tmpDir,
 		yamlDir:    filepath.Join(tmpDir, "yaml"),
-		Istioctl:   NewIstioctl(tmpDir, *namespace, *managerHub, *managerTag),
+		Istioctl:   istioctl,
```
Comment (sebastienvas) [inline tests/e2e/framework/kubernetes.go]: Those are not being used by kubernetes we can just add them in the main TestConfig. BUt let s do that in another PR because I need to merge to master.
